### ISSUE TITLE
NES Classic Mini: system does not reboot after rolling back to original kernel

### ISSUE THREAD

**Issue by Miva1** [TeamShinkansen/Hakchi2-CE#392]
Hi,

I have a Hakchi CE v.3.8.0 installed on my NES Classic Mini. I was trying to uninstall hakchi and roll back to original kernel but the console is not rebooting after the process is completed (original kernel installed). The console turns on but shows a message "shutting down" after few seconds on TV screen. 

I have tried it several times by reinstalling hakchi and then pressing the "Uninstall" button in Kernel-menu but it does not reboot with original kernel. If I reinstall hakchi, it works well, no problems.

Any ideas why the console is not willing to reboot after original kernel has been installed and what to try next?

**Comment by DanTheMan827** [comment]
I’m assuming you used hakchi2 not hakchi2 ce

you flashed the wrong kernel version back to the system when you uninstalled, see the instructions below:

So, you used Hakchi2.30, messed up your console, and now it just shuts down right away... and of course you want to fix it, right?

Well, depending on which system you have, you'll have to find one of these clean kernel backups from your favorite search engine (hint: search for the filename in quotes)

**Super Nintendo / Super Famicom Classic**
- kernel-dp-shvc-release-v2.0.12-0-gbff4fb3.img
- kernel-dp-shvc-release-v2.0.14-0-gd8b65c6.img
- kernel-dp-sneseur-release-v2.0.13-0-g9dca6c5.img
- kernel-dp-sneseur-release-v2.0.14-0-gd8b65c6.img
- kernel-dp-sneseur-release-v2.0.7-0-geb2b275.img
- kernel-dp-snesusa-release-v2.0.13-0-g9dca6c5.img
- kernel-dp-snesusa-release-v2.0.14-0-gd8b65c6.img
- kernel-dp-snesusa-release-v2.0.7-0-geb2b275.img

**Famicom Classic**
- kernel-dp-hvc-release-v1.0.5-0-g2f04d11.img
- kernel-dp-hvc-release-v1.0.8-0-g32708cb.img

**NES Classic (2016 Release)**
- kernel-dp-nes-release-v1.0.2-0-g99e37e1.img
- kernel-dp-nes-release-v1.0.3-0-gc4c703b.img

**NES Classic (2018 Release)**
- kernel-dp-nes-release-v1.0.7-0-g4ea4041.img

**Shonen Jump Special Edition Famicom Classic**
- kernel-dp-hvcj-release-v3.0.1-0-gad315e1.img

Then after you have the proper one for your system, you're going to need to download the latest release of Hakchi2 CE from <http://github.com/teamshinkansen/hakchi2/releases> and extract it to a nice and clean folder.

Enter FEL mode and select `Kernel > Uninstall` **while holding shift**, this will prompt you for a clean kernel dump (see above).

If you do not know what FEL Mode is, type !fel in main chat for a better explanation.

Hakchi should be removed from your system and it will start normally.

At this point, you can re-install hakchi using Hakchi2 CE and everything will work normally.

**Comment by Miva1** [comment]
Thanks for reply and instructions.
I actually used Hakchi CE 3.7.0 originally, around 1 year ago, when I installed custom kernel. I just happened to loose the original kernel and now struggling to roll back. 

I am not sure which production batch (1st or 2nd) my console is or which kernel version it should have. Is there a way to find it out except trial & error by following the procedure described above and trying to flash all the kernel versions one after one to find the correct one?
